### PR TITLE
Add some unit testing

### PR DESCRIPTION
**Pull request by fayaaz**

### CODE REVIEW

File: zegami_sdk/test.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        self.password = 'passWord'
+        self.local_token_path = os.path.join(Path.home(), 'mock.zegami.token')
+        self.url = TestZegamiClientMock(username=self.username, password=self.password)
+
+    def tearDown(self):
+        os.remove(self.local_token_path)
+
+    @requests_mock.Mocker()
+    def test_get_token(self, m):
+        m.post('https://mockzegami.com/oauth/token/', json={'token': 'asdkjsajgfdjfsda'})
```
Comment (mariya-gfx) [inline zegami_sdk/test.py:70]: Not sure why we need to POST here too as it is done in setUp() (so it POST before each test)?

File: zegami_sdk/test.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    TOKEN_NAME = 'mock.zegami.token'
+    HOME = 'https://mockzegami.com'
+
+
+class TestSdkUtil(unittest.TestCase):
+
+    @requests_mock.Mocker()
+    def setUp(self, m):
+        m.post('https://mockzegami.com/oauth/token/', json={'token': 'asdkjsajgfdjfsda'})
+        m.get('https://mockzegami.com/oauth/userinfo/', json={'projects': []})
```
Comment (mariya-gfx) [inline zegami_sdk/test.py:58]: What is the purpose of getting the userinfo?

File: zegami_sdk/test.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    @requests_mock.Mocker()
+    def test_get_token(self, m):
+        m.post('https://mockzegami.com/oauth/token/', json={'token': 'asdkjsajgfdjfsda'})
+        token = util._get_token(self.url, username=self.username, password=self.password)
+        self.assertTrue(token)
+        self.assertNotEqual(os.path.getsize(self.local_token_path), 0)
+
+    def test_ensure_token_user_login(self):
+        with requests_mock.Mocker() as m:
+            m.post('https://mockzegami.com/oauth/token/', json={'token': 'asdkjsajgfdjfsda'})
```
Comment (mariya-gfx) [inline zegami_sdk/test.py:77]: Same here..

File: zegami_sdk/test.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+class TestHelper(unittest.TestCase):
+
+    def test_no_libmagic_guess_data_mimetype(self):
+        """Force import error and check fallback mimetype set."""
+        test_output = io.StringIO()
+        sys.stdout = test_output
+        sys.modules.pop('magic')  # Unload the magic module
+        with patch('sys.path', []):  # Reload it with empty path
+            self.assertEqual(
+                guess_data_mimetype('mock_data'),
```
Comment (mariya-gfx) [inline zegami_sdk/test.py:31]: Where does 'mock_data' value come from?

**Review by mariya-gfx — APPROVED** [review]
Looks excellent!

File: zegami_sdk/test.py (hunk 1 above)
Comment (fayaaz) [reply zegami_sdk/test.py:70]:
```
self.url = TestZegamiClientMock(username=self.username, password=self.password)
```
The culprit is this line - it requires there to be a mock URL when instantiating the client

File: zegami_sdk/test.py (hunk 2 above)
Comment (fayaaz) [reply zegami_sdk/test.py:58]: Same again as the client instantion expects to hit that URL

File: zegami_sdk/test.py (hunk 3 above)
Comment (fayaaz) [reply zegami_sdk/test.py:77]: We need this for the next line to be mocked

**Pull request merged**